### ISSUE TITLE
Some suggestions from free hand drawing tools

### ISSUE THREAD

**Issue by EduCollin** [JamesLMilner/terra-draw#16]
Hi, I'm not a programmer, just someone that would like to use mapbox to teach geography and history live, using the program as a blackboard. For this, the plan is to have access to cool set of drawing tools.  This is why i feel very proud and hopeful of this kind of deep technical work you're undergoing!

Ideally, as a feedback from a layman, i would hope to have a few types of freehand drawing:

- Marker style, with opacity and thickness options, in which strokes applied over each other make the color more opaque (like https://www.scribblemaps.com/create does it)
- Highlighter style, with opacity and thickness options, in which a stroke overriding an earlier one doesn't tint the area more (so you don't ruin the coloring of an area just because you left a small gap undrawn )
- A fill in button with intensity parameter, like in traditional drawing programs.
- The already present freehand polygon is also cool! Maybe an option of modifying the shape of its vertices after the polygon is placed would be nice too, in maybe the color andthickness of the border too.
- Advanced tool: LAMPS. Is it possible to tinker with the atmospheric light of programs such as mapbox, so as that one can place light sources, like in a videogame engine like unity? I have no idea but i still write the idea here just in case.

Thank you for listening!!

Edu from Barcelona

**Comment by JamesLMilner** [comment]
Hi @EduCollin - thanks so much for your feedback, it's great to hear from you and your kind comments about the project!

I assume in your case you are talking about terradraw.io? In which case I do have plans to keep improving and adding useful features to the site.

Overall I agree with your first three points, it would be fun/useful to be able to add styles to the different tools to control the look and feel, maybe to also be able to style individual features although this may be harder. I will look into this more over time :) 

Regarding editing hand draw polygon, this is possible, I have just disabled it on the website. I could look at turning this on, or being able to control settings.

The lamps idea may be difficult as the website uses LeafletJS which I don't believe has such features. Although this may be possible in things like MapboxGL. I will be honest in saying this may be a bit outside the scope of the terradraw website.

Again thanks for your feedback! If you are looking for more 'comprehensive'  tools and premade map drawing product maybe take a look at https://www.placemark.io or perhaps https://www.felt.com :)

**Comment by EduCollin** [comment]
Thanks to you for reaching back.
Personally, my objective, in relation to the geography divulgation youtube project, is to be able to paint on a 3D map. So, i would use scribble maps but it's only 2D so i can't, and i'm waiting on the possibility that mapbox implements these tools. Felt and placemark are also 2D and only support line drawing, not brush painting. I wrote this issue here (and on mapbox's git) because i saw that your objective is to provide a cross platform architecture for drawing tools and wanted to make sure that tools i suggested are taken into consideration :)

**Comment by JamesLMilner** [comment]
@EduCollin apologies I meant to reply to your follow up here! I would need to test to see if this works in 3D - I think on Mapbox this should 'just work' but I'd need to double check. I will get back to you with an update!

**Comment by JamesLMilner** [comment]
Hey @EduCollin - I am going to close this but I am keeping track of the ability to draw on Globes via #23. Feel free to open a new issue if you have any more questions or suggestions.

**Comment by kuanb** [comment]
Hope it is ok to piggy back off this - I see the freehand class is an extension of `FreehandPolygon` - I would suggest option to have a `LineString` mode as well for those who want to "scribble" rather than "lasso".

**Comment by JamesLMilner** [comment]
@kuanb I've made a seperate issue for this :) #36
